shaders/src/light_indirect.fs

```diff
@@ -676,32 +672,30 @@ void applyRefraction(
     float lod = max(0.0, 2.0 * log2(tweakedPerceptualRoughness) + frameUniforms.refractionLodOffset);
 
     vec4 Fat = textureLod(light_ssr, p.xy, lod);
-    float at = Fat.a;
-#if defined(IS_TRANSPARENT_OR_REFRACTIVE)
-    vec3 Ft = mix(vec3(1.0), Fat.rgb, at);
-#else
-    vec3 Ft = Fat.rgb;
-#endif
 
+    // if it's a transparent background, shift color components towards white
+    vec3 Ft = mix(vec3(1.0), Fat.rgb, Fat.a);
+    float at = Fat.a;
 #endif
 
     // base color changes the amount of light passing through the boundary
     Ft *= pixel.diffuseColor;
+    at += (1.0 - at) * (1.0 - luminance(pixel.diffuseColor.rgb));
```

Comment:
It may be argued if it's worth refactoring this to
```
float oneMinusAt = 1.0 - at;
at += oneMinusAt - oneMinusAt * luminance(pixel.diffuseColor.rgb);
```
to ensure the FMA but it's not worth doing for now.